docs/siteconfig.rst

```diff
@@ -258,3 +258,40 @@ Here is a sample configuration with many of the options set and documented::
     # Do not allow more than that many jobs in a single run by default.
     # To disable this check use 0.
     job_threshold: 500
+
+Known Failure Patterns File Format
+==================================
+
+When using the ``--skip-known-failures`` option with ``teuthology-suite --rerun``,
+teuthology can filter out jobs with known failure patterns. The known patterns are
+loaded from a JSON or YAML file specified with ``--known-patterns-file``, or from
+the default bundled file ``teuthology/suite/patterns/known-failures.json``.
+
+The file format supports both JSON and YAML. The file must contain a ``patterns``
+key with a list of regex patterns (strings) that will be matched against job
+failure reasons.
+
+JSON format example::
+
+    {
+        "patterns": [
+            "Command failed on .* with status 1:",
+            "clocks not synchronized",
+            "cluster \\[WRN\\] Health check failed:.*OBJECT_UNFOUND"
+        ]
+    }
+
+YAML format example::
+
+    patterns:
+      - "Command failed on .* with status 1:"
+      - "clocks not synchronized"
+      - "cluster \\[WRN\\] Health check failed:.*OBJECT_UNFOUND"
+
+Patterns are matched using Python's ``re.search()`` function, so they support
+full regular expression syntax. If a job's ``failure_reason`` matches any of
+the patterns, it is considered a known failure and will be skipped during
+rerun when ``--skip-known-failures`` is enabled.
+
+Only jobs with failure reasons that don't match any known pattern will be
+scheduled for rerun.
```

Comment:
Thanks for adding this, but siteconfig is only for description of teuthology.yaml config file format hints. We need to find better place or dedicate a separate file.